Library/Homebrew/manpages/brew.1.md

```diff
@@ -345,12 +345,30 @@ Note that these flags should only appear after a command.
   * `switch` <name> <version>:
     Symlink all of the specific <version> of <name>'s install to Homebrew prefix.
 
-  * `tap` [--full] [<tap>]:
-    Tap a new formula repository from GitHub, or list existing taps.
-
-    <tap> is of the form <user>/<repo>, e.g. `brew tap homebrew/dupes`.
-
-    If `--full` is passed, a full clone will be used.
+  * `tap` [--full] [<name>] [<target>]:
+    Tap a new formula repository or list existing taps. This command can be
+    invoked in three ways. In what follows, <name> stands for something of the
+    form <user>/<repo>, e.g. `brew tap homebrew/dupes`.
+
+    + `tap` without arguments displays existing taps.
+    + `tap <name>` taps a new formula repository from GitHub using HTTP
+    + `tap <name> <target>` taps a new formula repository from anywhere
+
+    The one- and two-argument forms of `tap` make different assumptions and
+    have different advantages. Many taps are hosted on GitHub, and in order to
+    make things easier, `tap <name>` attempts to clone a repository at
+    https://github.com/#{user}/homebrew-#{repo}. This helps users in the
+    default situation, but it also limits the one-argument invocation to
+    GitHub. By contrast, `tap <name> <target>` does not make any assumptions
+    and it's more flexible, but the user must give an extra argument. The
+    two-argument form allows users to tap repositories from any location, using
+    any protocol that git recognizes.
```

Comment:
Might be good to slim this down a bit and move from prose to just an explanation of each form.